### PR TITLE
fix: issues in 'updateNewTeamMemberEventTypes' while adding new team member.

### LINKED ISSUE #17496: [CAL-4678] Assignment: "Add all team members, including future members" needs retoggle sometimes
CleanShot 2024-11-05 at 18 27 46@2x

this toggle every now and then doenst pick up new members (which is the whole point).

and when that happens, the user has to toggle off and on again to add the new people 

<sub>[CAL-4678](https://linear.app/calcom/issue/CAL-4678/assignment-add-all-team-members-including-future-members-needs)</sub>

### PR DESCRIPTION
**Pull request by vijayraghav-io**
## What does this PR do?

https://www.loom.com/share/fecc1e1fc21b42f78f632fe44809f92a?sid=8b5e3c42-8c7f-471c-b0ee-2174c90bea23

After fix:
https://www.loom.com/share/16d7af3a31ba46e2869f007c3ee04ef9?sid=3959f03c-8b61-4ace-971f-b58a4531fedd

## Root Cause
Zod error is occuring at this line when parent eventType is Managed.
It was failing silently, i.e. though membership was accepted, the eventTypes were not created for newly invited member.
https://github.com/calcom/cal.com/blob/3f7e632d5cc6b17781f866071e09c65e1827ba1c/packages/lib/server/queries/teams/index.ts#L430

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] - N/A - I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

### CODE REVIEW

**Comment by vijayraghav-io** [comment]
fixing type checks

**Comment by vijayraghav-io** [comment]
This was already handled in  "handleChildrenEventTypes.ts" , hence was working if re-toggled from UI.

https://github.com/calcom/cal.com/blob/8f92c9b884715b670254458ddc379931ab63ad50/packages/features/ee/managed-event-types/lib/handleChildrenEventTypes.ts#L117

**Comment by vijayraghav-io** [comment]
will try to add test by tomorrow

**Review by Udit-takkar — APPROVED** [review]
LGTM. 

Btw the issue you linked https://github.com/calcom/cal.com/issues/17496 is a different one. users didn't got any error while accepting team invite or sign up. that issue was fixed with this PR.

**Comment by vijayraghav-io** [comment]
Yes, the membership was accepted and updation or creation of eventTypes was failing silently, as shown in description.

File: packages/lib/server/queries/teams/index.ts [hunk 1]
```diff
@@ -454,7 +456,7 @@ export async function updateNewTeamMemberEventTypes(userId: number, teamId: numb
               users: {
                 connect: [{ id: userId }],
               },
-              parentId: eventType.parentId,
+              parentId: eventType.id,
```
[the remark this answers is not in the document]
Comment (vijayraghav-io) [reply packages/lib/server/queries/teams/index.ts:459]: manually re-toggling "assignAllTeamMembers" in assignment Tab, was reconnecting the children.

**Comment by emrysal** [comment]
@vijayraghav-io The issue with https://github.com/calcom/cal.com/issues/17496 was that through SAML signup the updating and creating of event types wasn't triggered at all; which was fixed in that PR. This builds on that further; happy to merge btw 🙇

**Pull request merged**